packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -979,9 +974,10 @@ function performSyncWorkOnRoot(root) {
 
   // We now have a consistent tree. Because this is a sync render, we
   // will commit it even if something suspended.
-  root.finishedWork = (root.current.alternate: any);
+  const finishedWork: Fiber = (root.current.alternate: any);
+  root.finishedWork = finishedWork;
   root.finishedExpirationTime = expirationTime;
-
+  root.nextKnownPendingLevel = getRemainingExpirationTime(finishedWork);
```

Comment:
Is it worth write a regression test that would fail if this *and* the defensive safeguard were removed? Looks like the new test covers the Concurrent render only. Not sure if this matters.

Reply:
The diff view is misleading because of collapsed sections. This line is the sync path. The concurrent path is here: https://github.com/facebook/react/pull/18535/files#diff-63c455475b8566c10993b2daa2c3211bR685

The test #18515 covers the sync path because the discrete update is expired but I can add another test since that one happens to be fixed by the other change.